src/Recursor.h

```diff
@@ -611,17 +611,17 @@ void Recursor<Derived>::ExtendBeta(const M& beta, size_t lastColumn, M& ext, int
         std::tie(beginRow, endRow) = beta.UsedRowRange(lastColumn + 1);
     else
         std::tie(beginRow, endRow) = beta.UsedRowRange(lastColumn);
-    for (int j = 0; j <= lastColumn && j <= numExtColumns; ++j)
+    for (size_t j = 0; j <= lastColumn && j <= numExtColumns; ++j)
         beginRow = std::min(static_cast<int>(beta.UsedRowRange(lastColumn - j).first), beginRow);
 
-    for (int j = lastColumn; j + numExtColumns > lastColumn; j--) {
+    for (int j = lastColumn; j + numExtColumns - lastColumn > 0; j--) {
         /* Convert from old template to new template coordinates.
            lengthDiff will be 0 for substitution, -1 for deletion and +1 for
            insertion
          */
-        int jp = j + lengthDiff;
+        size_t jp = j + lengthDiff;
         // What is the current extension column we are adding data into.
-        int extCol = lastExtColumn - (lastColumn - j);
+        size_t extCol = lastExtColumn - (lastColumn - j);
```

Comment:
Oh shit. Here we go again.

extCol = lastExtColumn - (lastColumn - j) =>
extCol = lastExtColumn - lastColumn + j
    where: lastExtColumn = numExtColumns - 1 and
    where: numExtColumns = 1 + lengthDiff + lastColumn =>
extCol = (numExtColumns - 1) - lastColumn + j =>
extCol = 1 + lengthDiff + lastColumn - 1 - lastColumn + j =>
extCol = j + lengthDiff =>

and previously: j + lengthDiff >= 0

So we're okay.

But I'm seeing that extCol = jp. Am I missing something?